subsys/logging/log_core.c

```diff
@@ -539,19 +539,20 @@ bool z_impl_log_process(void)
 	if (IS_ENABLED(CONFIG_LOG_MODE_DEFERRED)) {
 		bool dropped_pend = z_log_dropped_pending();
 		bool unordered_pend = z_log_unordered_pending();
+		uint64_t now = k_uptime_get();
 
-		if ((dropped_pend || unordered_pend) &&
-		   (k_uptime_get() - last_failure_report) > CONFIG_LOG_FAILURE_REPORT_PERIOD) {
+		if ((dropped_pend || unordered_pend) && (now > last_failure_report) &&
```

Comment:
do we need this `now > last failure_report` if it is only for testing purposes? without calling `sys_clock_tick_set()` and `set_log_last_failure_report_tick()` it will never happen.